feat: parallel unit test runners (#4395)
FuelLabs/sway: docs/book/src/testing/unit-testing.md

```toml
{{#include ../../../../examples/multi_contract_calls/caller/Forc.toml:multi_contract_call_toml}}
```

## Running Tests in Parallel or Serially

> By default, tests are run in parallel using threads. Note that each test is assigned their own storage so different tests does not invalidate a shared state by running in parallel.

Reviewer: ```suggestion
By default, all unit tests in your project are run in parallel. Note that this does not lead to any data races in storage because each unit test has its own storage space that is not shared by any other unit test.
```

> When you run multiple tests, by default they run in parallel using threads, meaning they finish running faster and you get feedback quicker. It is important to note that each test is assigned their own storage so different tests does not invalidate a shared state by running in parallel.

Reviewer 2: ```suggestion
By default, tests are run in parallel using threads. Note that each test is assigned their own storage so different tests does not invalidate a shared state by running in parallel.
```

> `forc` will use necessary number of threads available in your system by default, but it is possible to specify number of threads to utilize by using `--test-threads` flag.

Reviewer: ```suggestion
By default, `forc test` will use all the available threads in your system. To request that a specific number of threads be used, the flag `--test-threads <val>` can be provided to `forc test`.
```

> By default `forc` will use necessary number of threads available in your system but it is possible to specify number of threads to utilize by using `--test-threads` flag.

Reviewer 2: ```suggestion
`forc` will use necessary number of threads available in your system by default, but it is possible to specify number of threads to utilize by using `--test-threads` flag.
```

Looks nicer to not start with 'By default' on this line if we use it above as well
